index.js

```diff
@@ -113,7 +125,13 @@ module.exports = (bookshelf, settings) => {
             let field = get(this, 'bcrypt.field')
 
             if (model.hasChanged(field) && options.bcrypt !== false) {
-              return hashAndStore(model.get(field), field, model)
+              let password = model.get(field)
+
+              if (password !== null && password !== undefined) {
```

Comment:
`&& typeof password !== 'undefined'` to get rid of the codacy error

can read more here: http://eslint.org/docs/2.0.0/rules/no-undefined